dffml/source/mysql_source.py

```diff
@@ -0,0 +1,114 @@
+import json
+import aiomysql
+from typing import AsyncIterator, NamedTuple, Dict
+
+from dffml.base import BaseConfig
+from dffml.repo import Repo
+from dffml.source.source import BaseSourceContext, BaseSource
+from dffml.util.cli.arg import Arg
+from dffml.util.entrypoint import entry_point
+
+
+class MysqlSourceConfig(BaseConfig, NamedTuple):
+    host: str
+    port: int
+    user: str
+    password: str
+    db: str
+    query: str
+
+
+class MysqlSourceContext(BaseSourceContext):
+    async def update(self, repo: Repo):
+        db = self.conn
+        marshall = json.dumps(repo.dict())
```

Comment:
Ah, we probably want to add another parameter to `MysqlSourceConfig`, which will be the order of things to update, as given in the update query. So for an update query such as:

```sql
INSERT INTO source_data (`key`, `column_a`, `column_b`) VALUES(%s, %s, %s) ON DUPLICATE KEY UPDATE column_a = %s, column_b = %s;
```

We will need a parameter which is the feature names in order.

```console
-model-features feature_a feature_b
```

Which maps to the columns they are stored in, `column_a` and `column_b`.  You'll probably need an `collections.OrderedDict`.